flask/views.py

```diff
@@ -102,12 +102,16 @@ def view(*args, **kwargs):
         return view
 
 
+def get_methods(cls):
+    return getattr(cls, 'methods', []) or []
+
+
 class MethodViewType(type):
 
     def __new__(cls, name, bases, d):
         rv = type.__new__(cls, name, bases, d)
         if 'methods' not in d:
-            methods = set(rv.methods or [])
+            methods = set(m for b in bases for m in get_methods(b))
```

Comment:
I'm not sure to understand. Let's assume this deep inheritance:

```python
class MaternalGrandMother(MethodView):
    def get(self):
        # implementation details

class MaternalGrandFather(MethodView):
    def post(self):
        # implementation details

class Mother(MaternalGrandMother, MaternalGrandFather):
    # method = ['POST', 'GET']

class PaternalGrandMother(MethodView):
    def delete(self):
        # implementation details
    def patch(self):
        # implementation details

class PaternalGrandFather(MethodView):
    def put(self):
        # implementation

class Father(PaternalGrandMother, PaternalGrandFather):
    methods = ['PUT', 'DELETE']  # Don't inherit PATCH
```

So now here's the behavior I expect, and I don't think, and that's what I meant in this code:

```python
class A(Father, Mother):
    # methods = ['PUT', 'DELETE', 'GET', 'POST']

class B(Mother, PaternalGrandFather, PaternalGrandMother):
    # methods = ['PUT', 'DELETE', 'PATCH', 'GET', 'POST']

class C(Mother, MaternalGrandMother, PaternalGrandMother):
    # methods = ['POST', 'GET', 'DELETE', 'PATCH']
```

Can you give me an example that I'm missing?